ctrl / command + click open new window not working anymore.

Hi guys, I have noticed that ctrl / command + click open new window not working anymore on a link with uisref on it. Were there any changes to the behaviour? this is such an important and common thing

I have been looking into this issue and it should be possible to solve in the same way Angular's router does it. I've been trying to put together a PR, but when I follow the instructions to get the project up and running npm install fails inside the angular package.

```npm ERR! path /Users/euan/Sites/uirouter/angular/node_modules/@uirouter/publish-scripts/update_changelog.js
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod
npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/euan/Sites/uirouter/angular/node_modules/@uirouter/publish-scripts/update_changelog.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent ```

I can't find any way to work around this problem

--- a/src/directives/uiSref.ts
+++ b/src/directives/uiSref.ts
@@ -1,7 +1,7 @@
 /** @ng2api @module directives */
 /** */
 import { UIRouter, extend, Obj, TransitionOptions, TargetState } from "@uirouter/core";
-import { Directive, Inject, Input, Optional, ElementRef, Renderer2, OnChanges, SimpleChanges } from "@angular/core";
+import { Directive, Inject, Input, Optional, ElementRef, Renderer2, OnChanges, SimpleChanges, HostListener } from "@angular/core";
 import { UIView, ParentUIViewInject } from "./uiView";
 import { ReplaySubject } from "rxjs/ReplaySubject";
 import { Subscription } from "rxjs/Subscription";
@@ -69,7 +69,6 @@ export class AnchorUISref {
 @Directive({
   selector: '[uiSref]',
   exportAs: 'uiSref',
-  host: { '(click)': 'go()' }
 })
 export class UISref implements OnChanges {
 
@@ -169,9 +168,10 @@ export class UISref implements OnChanges {
   }
 
   /** When triggered by a (click) event, this function transitions to the UISref's target state */
-  go() {
-    if (this._anchorUISref && this._anchorUISref.openInNewTab()) {
-      return ;
+  @HostListener("click", ["$event.button", "$event.ctrlKey", "$event.metaKey"])
+  go(button: number, ctrlKey: boolean, metaKey: boolean) {
+    if (this._anchorUISref && (this._anchorUISref.openInNewTab() || button !== 0 || ctrlKey || metaKey)) {
+      return;
     }
 
     this._router.stateService.go(this.state, this.params, this.getOptions());

--- a/test/uiSref/uiSref.spec.ts
+++ b/test/uiSref/uiSref.spec.ts
@@ -104,7 +104,7 @@ describe('uiSref', () => {
 
         describe('when clicked', () => {
           beforeEach(() => {
-            des[0].triggerEventHandler('click', {});
+            des[0].triggerEventHandler('click', {button: 0, metaKey: false, ctrlKey: false});
           });
 
           it('should ignore the click event', () => {
@@ -121,7 +121,7 @@ describe('uiSref', () => {
 
         describe('when clicked', () => {
           beforeEach(() => {
-            des[0].triggerEventHandler('click', {});
+            des[0].triggerEventHandler('click', {button: 0, metaKey: false, ctrlKey: false});
           });
 
           it('should navigate to the state', () => {

--- a/test/uiSrefStatus/uiSrefStatus.spec.ts
+++ b/test/uiSrefStatus/uiSrefStatus.spec.ts
@@ -39,7 +39,7 @@ describe('uiSrefStatus', () => {
   describe('when click on `foo` uiSref', () => {
     beforeEach(async(() => {
       spyOn(component, 'updated');
-      de.triggerEventHandler('click', {});
+      de.triggerEventHandler('click', {button: 0, metaKey: false, ctrlKey: false});
     }));
 
     it('should emit a event with a TargetState pointing to `foo`', () => {

